public/app/plugins/datasource/testdata/runStreams.ts

```diff
@@ -104,18 +102,13 @@ export function runSignalStream(
     }
 
     const pushNextEvent = () => {
-      addNextRow(Date.now());
-
-      const elapsed = liveTimer.lastUpdate - lastSent;
-      if (elapsed > 1000 || liveTimer.ok) {
-        subscriber.next({
-          data: [frame],
-          key: streamId,
-          state: LoadingState.Streaming,
-        });
-        lastSent = liveTimer.lastUpdate;
-      }
-
+      lastSent = Date.now();
+      addNextRow(lastSent);
+      subscriber.next({
```

Comment:
I understand that `liveTimer.ok` was returning always true but I wonder if this original condition is relevant and it was broken. Are you sure the elapsed time validation isn't needed anymore?

Reply:
yep, this is just executing `pushNextEvent` using `speed` as the period